app/components/post/post.js

```diff
@@ -263,7 +266,7 @@ export default class Post extends PureComponent {
         const style = getStyleSheet(theme);
         const isReplyPost = this.isReplyPost();
         const onUsernamePress =
-            Config.ExperimentalUsernamePressIsMention && !channelIsReadOnly ? this.autofillUserMention : this.viewUserProfile;
+            (Config.ExperimentalUsernamePressIsMention && !channelIsReadOnly && canPost) ? this.autofillUserMention : this.viewUserProfile;
```

Comment:
I'm inclined to remove `ExperimentalUsernamePressIsMention` this was added by shift and is experimental and only applies on build time. If we remove it we wouldn't need the  `canPost` or `channelIsReadOnly` props.

@esethna thoughts?

Reply:
@enahum @esethna Just wondering if it would be OK to decouple the above decision from this feature/PR?

Reply:
I'm 100% on this, but lets wait for @esethna input

Reply:
Agree on not blocking this PR on this decision. 

I'm not sure what "ExperimentalUsernamePressIsMention" is, but if it's an experimental config setting we can't just drop support for it without warning customers and pre-communicating

Reply:
@esethna no customer other than shift uses this AND you know what happened with shift, right?

won;t block this PR but I'm submitting another PR to get rid of this right now.

Reply:
After some discussion with @esethna I'll be submitting a PR to get rid of the feature. @mkraft you can revert the changes on this file for `canPost`